File: python3/404/Solution.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def sumOfLeftLeaves(self, root: Optional[TreeNode]) -> int:
        if root is None:
            return 0

        left_child = root.left
        right_child = root.right

        if left_child is not None and left_child.left is None and left_child.right is None:
            return left_child.val + self.sumOfLeftLeaves(right_child)

        return self.sumOfLeftLeaves(left_child) + self.sumOfLeftLeaves(right_child)

File: java/404/Solution.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public int sumOfLeftLeaves(TreeNode root) {
        if (root == null) {
            return 0;
        }

        TreeNode leftChild = root.left;
        TreeNode rightChild = root.right;

        if (leftChild != null
            && leftChild.left == null
            && leftChild.right == null) {
            return leftChild.val + this.sumOfLeftLeaves(rightChild);
        }

        return this.sumOfLeftLeaves(leftChild) + this.sumOfLeftLeaves(rightChild);
    }
}

